### PR TITLE
[TASK] Make the CssInliner inherit AbstractHtmlProcessor

### LINKED ISSUE #554: Make the CssInliner inherit AbstractHtmlProcessor

### PR DESCRIPTION
**Pull request by oliverklee**
Closes #554

### CODE REVIEW

File: src/Emogrifier/CssInliner.php [hunk 1]
```diff
@@ -275,59 +174,6 @@ private function getHeadElement()
...
-     * Creates a DOMDocument instance from the given HTML and stores it in $this->domDocument.
-     *
-     * @param string $html
-     *
-     * @return void
-     */
-    private function createRawDomDocument($html)
-    {
-        $domDocument = new \DOMDocument();
-        $domDocument->encoding = 'UTF-8';
```
Comment (JakeQZ) [inline src/Emogrifier/CssInliner.php]: `AbstractHtmlProcessor` does not set `encoding`.  Should it?

File: src/Emogrifier/CssInliner.php [hunk 2]
```diff
@@ -275,59 +174,6 @@ private function getHeadElement()
...
-        $domDocument->strictErrorChecking = false;
-        $domDocument->formatOutput = true;
-        $libXmlState = \libxml_use_internal_errors(true);
-        $domDocument->loadHTML($this->prepareHtmlForDomConversion($html));
-        \libxml_clear_errors();
-        \libxml_use_internal_errors($libXmlState);
-        $domDocument->normalizeDocument();
```
Comment (JakeQZ) [inline src/Emogrifier/CssInliner.php]: `AbstractHtmlProcessor` does not call `normalizeDocument()`.  Should it?

**Comment by oliverklee** [comment]
> We should also remove the `CssInliner` tests that duplicate those for `AbstractHtmlProcessor` but this can be a separate PR.

I agree. In the separate PR, I'll also add a test that `CssInliner` extends `AbstractHtmlProcessor`.

File: src/Emogrifier/CssInliner.php (hunk 1 above)
Comment (oliverklee) [reply src/Emogrifier/CssInliner.php]: Good catch! No, because `loadHTML` will overwrite the encoding anyway. I'll create a PR for removing that line.

File: src/Emogrifier/CssInliner.php (hunk 2 above)
Comment (oliverklee) [reply src/Emogrifier/CssInliner.php]: Probably. It looks like this is not covered with tests yet. I'll create a PR for that.

Comment (oliverklee) [reply src/Emogrifier/CssInliner.php]: I haven't been able to come up with a regression test for the missing `normalizeDocument`. As it's recommended to use it, I'll create a PR anyway.

Comment (oliverklee) [reply src/Emogrifier/CssInliner.php]: #662

File: src/Emogrifier/CssInliner.php (hunk 1 above)
Comment (oliverklee) [reply src/Emogrifier/CssInliner.php]: #661

**Comment by oliverklee** [comment]
Rebased and repushed.
